### DISCUSSION TITLE
Non standard cut behavior.

### DISCUSSION THREAD

**Discussion by Phoenixxl** [eclipse-theia/theia#13650] [Improvements] [answered]
An issue is still the non standard cut behavior compared to all other editors. 

When the option of cutting/copying the current line is disabled. ( editor: empty selection clipboard -> uncheck Controls whether copying without a selection copies the current line.)

In this mode:

Expected, "normal" behaviour -> 
Selecting a piece of text "abc", copying it to clipboard , pasting it, pastes copied text "abc". 
Selecting nothing. pressing the cut or copy shortcut while there is **no selection**, pasting , pastes the content of the clipboard which is still "abc".

Behavior seen in all thea based editors ->
Selecting a piece of text "abc", copying it to clipboard , pasting it, pastes copied text "abc". 
Selecting nothing. pressing the cut or copy shortcut while there is **no selection**, pasting , pastes the content of the clipboard which is still **nothing**.

This is an oversight, this is not intended dear forum moderator or equivalent I beg, don't use sophisms to try and make this go away , leave this open ideally for the person who wrote that bit to read this. 

When cutting or copying nothing a clipboard stays in it's current state it doesn't get emptied. In theia based editors it gets emptied.

**Comment by msujew** [comment, marked as answer]
Hey @Phoenixxl,

do you mind filing an issue for this? This works as expected in vscode, but incorrectly in Theia (even though we use the same editor), so we should adjust out behavior.

**Reply by Phoenixxl** [reply]
https://github.com/eclipse-theia/theia/issues/13652
